test/functional/mining_basic.py

```diff
@@ -159,6 +159,15 @@ def test_timewarp(self):
         bad_block.solve()
         assert_raises_rpc_error(-25, 'time-timewarp-attack', lambda: node.submitheader(hexdata=CBlockHeader(bad_block).serialize().hex()))
```

Comment:
I'm not immediately seeing why the previous check (for the `bad_block.nTime = t` case) is needed anymore now that the check below is being added (`bad_block.nTime = t + MAX_FUTURE_BLOCK_TIME - MAX_TIMEWARP - 1` case).  Maybe I'm overlooking something?

If not, then could add a commit that removes the check for the `bad_block.nTime = t` case.